### PR TITLE
COMPAT: tzawareness behavior to be same as datetime

### LINKED ISSUE #28507: COMPAT: tzawareness check behavior is different from stdlib
`Timestamp` comparisons raise TypeError in some cases where `datetime`s do not:

```
ts = pd.Timestamp.now()
ts2 = ts.tz_localize("UTC")
dt = ts.to_pydatetime()
dt2 = ts2.to_pydatetime()

>>> ts == ts2   # <-- TypeError
>>> dt == dt2
False
```

The datetime comparison _will_ raise for inequalities, just not `==` or `!=`.

Changing this behavior would be an API change, but it would likely simplify a bunch of headaches (including a bug in array_equivalent)

### PR DESCRIPTION
**Pull request by ShaharNaveh**
- [x] closes #28507
- [x] tests added / passed
- [x] passes `black pandas`
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

### CODE REVIEW

**Comment by MomIsBestFriend** [comment]
cc @jbrockmendel

File: pandas/_libs/tslibs/c_timestamp.pyx [hunk 1]
```diff
@@ -119,7 +119,9 @@ cdef class _Timestamp(datetime):
             else:
                 return NotImplemented
 
-        self._assert_tzawareness_compat(other)
+        if op not in [Py_EQ, Py_NE]:
+            self._assert_tzawareness_compat(other)
+
```
Comment (jbrockmendel) [inline pandas/_libs/tslibs/c_timestamp.pyx]: if we dont have tzawareness compat here, then eq should be False and ne should be True, i.e. dont just fall through to cmp_scalar here

**Comment by jbrockmendel** [comment]
BTW, i think once this is done we can get rid of some (all?) uses of comp_method_OBJECT_ARRAY

**Comment by WillAyd** [comment]
@MomIsBestFriend still active? Can you try to get CI passing?

**Comment by MomIsBestFriend** [comment]
@WillAyd I don't know if this is still active, I am not totally familiar with the code base, and this might been not the right issue to take, might be out of my level, I think I'll close this, due to the fact I can't really solve it. sorry for the waste of time.

**Pull request closed without merging**